pyleco/errors.py

```diff
@@ -0,0 +1,47 @@
+#
+# This file is part of the PyLECO package.
+#
+# Copyright (c) 2023-2023 PyLECO Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from typing import Any
+
+from jsonrpcobjects.objects import ErrorObject, ErrorObjectData, ErrorResponseObject
+
+# TODO define valid error codes
+
+# Routing errors (Coordinator)
+NOT_SIGNED_IN = ErrorObject(code=1234, message="You did not sign in!")
+DUPLICATE_NAME = ErrorObject(code=456, message="The name is already taken.")
```

Comment:
These error messages could end up being quite unhelpful, as they are not telling you _what value_ they are complaining about. Or is this handled in another way (in that `data` arg)? "The name 'XYZ' is already taken" is immensely more helpful. Similar for some others here.

Reply:
Exactly, the name will enter into `data`.